### PR TITLE
Vehicle damage script

### PR DESCRIPTION
**Pull request by Urbaniized**
Video takes too long to process and upload but I can show you in a live stream

Made sure that it doesn't affect any current systems.

Vehicle gets repairs at LS customs and repair shops
Vehicle stays orange if put in garage (doesn't reset engine health)
Mechanic repair kits work and count towards repairs for successful repairs

### CODE REVIEW

**Comment by Urbaniized** [comment]
forgot to add the new server CFG

just do `ensure usa_seatbelt`

**Comment by minipunch** [comment]
I'm confused

Why is there a new `usa_seatbelt` resource and why has it not been mentioned ??

You've only mentioned there is a new vehicle damage script which I assume is what the changes to usa_vehdamage is

**Review by minipunch — COMMENTED** [review]
see comments

**Comment by Urbaniized** [comment]
The current one basically starts you at 100 health then moves to 50 health then 0 health.

This one work by slowly going from 100-0. If you hit a stop sign, it will damage the engine but it won't instantly go to 50 health. The harder you drive the car with a damaged engine the faster the engine degrades.

https://github.com/StockholmCityRP/fivem-realisticvehicle

It was the only one I could find that will work with the current framework. Also it is highly customizable vs what we have right now.

File: resources/usa_vehdamage/config.lua [hunk 1]
```diff
@@ -0,0 +1,62 @@
+
+-- IMPORTANT: Some of these values MUST be defined as a floating point number. ie. 10.0 instead of 10
+Config = {
+	deformationMultiplier = -1,					-- How much should the vehicle visually deform from a collision. Range 0.0 to 10.0 Where 0.0 is no deformation and 10.0 is 10x deformation. -1 = Don't touch. Visual damage does not sync well to other players.
+	deformationExponent = 0.4,					-- How much should the handling file deformation setting be compressed toward 1.0. (Make cars more similar). A value of 1=no change. Lower values will compress more, values above 1 it will expand. Dont set to zero or negative.
+	collisionDamageExponent = 0.6,				-- How much should the handling file deformation setting be compressed toward 1.0. (Make cars more similar). A value of 1=no change. Lower values will compress more, values above 1 it will expand. Dont set to zero or negative.
+
+	damageFactorEngine = 1.0,					-- Sane values are 1 to 100. Higher values means more damage to vehicle. A good starting point is 10
```
Comment (minipunch) [inline resources/usa_vehdamage/config.lua:8]:
```suggestion
	damageFactorEngine = 2.0,					-- Sane values are 1 to 100. Higher values means more damage to vehicle. A good starting point is 10

```

File: resources/usa_vehdamage/config.lua [hunk 2]
```diff
@@ -0,0 +1,62 @@
+
+-- IMPORTANT: Some of these values MUST be defined as a floating point number. ie. 10.0 instead of 10
+Config = {
+	deformationMultiplier = -1,					-- How much should the vehicle visually deform from a collision. Range 0.0 to 10.0 Where 0.0 is no deformation and 10.0 is 10x deformation. -1 = Don't touch. Visual damage does not sync well to other players.
+	deformationExponent = 0.4,					-- How much should the handling file deformation setting be compressed toward 1.0. (Make cars more similar). A value of 1=no change. Lower values will compress more, values above 1 it will expand. Dont set to zero or negative.
+	collisionDamageExponent = 0.6,				-- How much should the handling file deformation setting be compressed toward 1.0. (Make cars more similar). A value of 1=no change. Lower values will compress more, values above 1 it will expand. Dont set to zero or negative.
+
+	damageFactorEngine = 1.0,					-- Sane values are 1 to 100. Higher values means more damage to vehicle. A good starting point is 10
+	damageFactorBody = 1.5,						-- Sane values are 1 to 100. Higher values means more damage to vehicle. A good starting point is 10
```
Comment (minipunch) [inline resources/usa_vehdamage/config.lua:9]:
```suggestion
	damageFactorBody = 2.5,						-- Sane values are 1 to 100. Higher values means more damage to vehicle. A good starting point is 10

```

**Review by minipunch — CHANGES_REQUESTED** [review]

**Review by minipunch — APPROVED** [review]

**Pull request merged**